.github/workflows/codspeed.yml

```diff
@@ -0,0 +1,59 @@
+name: CodSpeed Benchmarks
+
+on:
+  pull_request:
+    paths-ignore:
+      - 'docs/**'
+      - 'justfile'
+      - '.devcontainer'
+      - '**/*.md'
+      - '.gitignore'
+  merge_group:
+  push:
+    branches:
+      - main
+
+concurrency:
+  group: codspeed.yml-${{ github.event.pull_request.number || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  benchmarks:
+    name: Run benchmarks
+    runs-on: ubuntu-22.04
```

Comment:
@danlapid can we use large runners for this?

Reply:
Not sure how large runners work with github enterprise.
Perhaps @fhanau or @penalosa know.
In general I'd say why not use large runners for everything if they exist and work well.

Reply:
Important question, but this is beyond the IC level and depends on our GitHub plan (is there a certain amount of paid runner usage included with our plan, or do we decide to pay for a given amount of usage, based on that making larger runners available needs to be decided on the org level). IMHO the most impactful change would be to use paid runners for Windows, but other jobs could benefit too.